homeassistant/components/tplink/switch.py

```diff
@@ -37,10 +37,51 @@ async def async_setup_entry(
             entities.append(SmartPlugSwitch(child, coordinator))
     else:
         entities.append(SmartPlugSwitch(device, coordinator))
+        entities.append(SmartPlugSwitch(SmartDeviceLedWrapper(device), coordinator))
 
     async_add_entities(entities)
 
 
+class SmartDeviceLedWrapper(SmartDevice):  # type: ignore
```

Comment:
Please update the library instead. We don't accept code that should be part of the library.

Reply:
Hello,

This shouldn't be part of the library since the library offers options on the switch directly to turn on and off the led. 

hat being said, if that is something that could be usable directly in Home Assistant, I would love some pointers. With my investigations, I only found using that wrapper as a solution, here, which wouldn't make sense in the library because the option is provided directly on the switch itself.

Would really love some pointers to make this work!

Reply:
If the device already has support, why do we need to wrap it? I'd suggest just creating a new entity class that uses the correct device methods.